### DISCUSSION TITLE
onInit auto scroll if I know item id

### DISCUSSION THREAD

**Discussion by IvanHoliak** [asmyshlyaev177/react-horizontal-scrolling-menu#295] [Q&A] [answered]
If there is an item id in the url search params when you first visit the page, is it possible to make auto scroll directly to this item?

like this

`
<ScrollMenu
  itemClassName='mr-2 last:mr-0'
  onMouseDown={onMouseDown}
  onMouseMove={handleDrag}
  onMouseUp={onMouseUp}
  onWheel={onWheel}
  onInit={context => {
   if(itemId) context.scrollToItem(context.getItemById(itemId))
  }}
>
  {items
    .map((item) => (
`

**Comment by asmyshlyaev177** [comment, marked as answer]
@IvanHoliak I think so, is this example is what you looking for ? https://asmyshlyaev177.github.io/react-horizontal-scrolling-menu/?path=/story/examples-scrolltoitem--scroll-to-item

**Comment by IvanHoliak** [comment]
@asmyshlyaev177 
but if I use `apiRef` and `useEffect`
then `apiRef.current.getItemById` will return me `undefined`

in the example I see a hack using `setTimeout`
and then the answer will be expected, not undefined

**Reply by asmyshlyaev177** [reply]
Can also try `onInit` and `onUpdate` callbacks.

**Reply by IvanHoliak** [reply]
with `onInit` same as with `apiRef`
items size is zero on first render and `getItemById` will return `undefined`

in my case `onUpdate` is not suitable

that's why the solution with `setTimeout` looks promising and I'm already using it

thanks for the quick response and advice!

**Reply by IvanHoliak** [reply]
Ideally, it would be better to do all this in `onInit` when the component is first rendered

**Reply by asmyshlyaev177** [reply]
I will try to fix it when have a time.
